### ISSUE TITLE
Add Pixi Build Support

### ISSUE THREAD

**Issue by saulshanabrook** [readthedocs/readthedocs.org#11548]
## What's the problem this feature will solve?

I am switching to [pixi](https://pixi.sh/latest/) and would like to use it to build my Sphinx docs in the RTD. It provides lockfiles and allows conda + pip installations of packages, and uses the [uv](https://github.com/astral-sh/uv) resolver.

## Describe the solution you'd like

It would be great to be able to specify pixi like you can with conda or pip in the read the docs config.

## Alternative solutions

There is an alternative example posted in the Pixi docs (https://github.com/prefix-dev/pixi/pull/1423), but it requires installing Pixi through conda and then running a custom command to build the docs using pixi.

**Comment by humitos** [comment]
(quick reply from the phone) 

You can use `build.commands` to install pixi and use it, see https://docs.readthedocs.io/en/latest/config-file/v2.html#build-commands for more info about `build.commands`

**Comment by akhmerov** [comment]
`build.commands` seems to not be compatible with reusing existing tools for sphinx or mkdocs. I believe the idea of this issue is to only have the environment provided by pixi without needing to replicate the rest of the build process.

**Comment by humitos** [comment]
I understand. We don't currently support that use case. Also, from past discussions I can say we won't implement this "natively" and expose a configuration option like [`python.install`](https://docs.readthedocs.io/en/stable/config-file/v2.html#python-install) that uses `pixi` or `uv` or `poetry` or any other package manager.

Instead, we will be working on a way to override [pre-defined jobs](https://docs.readthedocs.io/en/stable/build-customization.html#extend-the-build-process), like `build.jobs.install`, and support this use case in a more generic way that will allow you to use any package manager:

```yaml
build:
  jobs:
    install:
      - pixi global install starship
      - pixi global install ruff
```

_(read more about this idea in [this other GitHub comment](https://github.com/readthedocs/readthedocs.org/issues/11216#issuecomment-1998186335))_

As a workaround until we implement this, I recommend you to follow similar steps than Poetry users: https://docs.readthedocs.io/en/stable/build-customization.html#install-dependencies-with-poetry or something around those lines. If you find a way to do this that you are happy with, feel free to open a PR to update that documentation page.

**Comment by pavelzw** [comment]
with pixi now having an asdf plugin, we could just install it from there

see https://github.com/prefix-dev/pixi/issues/1356#issuecomment-2169478022

**Comment by humitos** [comment]
> Instead, we will be working on a way to override [pre-defined jobs](https://docs.readthedocs.io/en/stable/build-customization.html#extend-the-build-process), like `build.jobs.install`, and support this use case in a more generic way that will allow you to use any package manager:

This was implemented in #11551, so users are able to use `build.jobs.install` 👍🏼
